pkg/aggregator/aggregator.go

```diff
@@ -156,6 +156,22 @@ func (s *referenceWalker) Start() {
 
 // FilterSpecByPaths remove unnecessary paths and unused definitions.
 func FilterSpecByPaths(sp *spec.Swagger, keepPathPrefixes []string) {
+	// Walk all references to find all unused definitions. This function
+	// want to only deal with unused definitions resulted from filtering paths.
+	initialUnusedDefinitions := map[string]bool{}
+	for k := range sp.Definitions {
+		initialUnusedDefinitions[k] = true
+	}
+	walkOnAllReferences(func(ref spec.Ref) spec.Ref {
+		if ref.String() != "" {
+			refStr := ref.String()
+			if strings.HasPrefix(refStr, definitionPrefix) {
+				delete(initialUnusedDefinitions, refStr[len(definitionPrefix):])
```

Comment:
turn this around: initialUnusedDefinitions -> initiallyUsedDefinitions. Then you can get rid of the first loop.

Reply:
Fixed.